packages/aws-cdk-lib/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,375 @@
+    this.version = '2021-06-01';
+
+    var findingsDestination: FindingsDestination = {};
+
+    if (props.logGroupNameAuditDestination) {
+      findingsDestination.cloudWatchLogs = {
+        logGroup: props.logGroupNameAuditDestination,
+      };
+    }
+
+    if (props.s3BucketNameAuditDestination) {
+      findingsDestination.s3 = {
+        bucket: props.s3BucketNameAuditDestination,
+      };
+    }
+
+    if (props.deliveryStreamNameAuditDestination) {
+      findingsDestination.firehose = {
+        deliveryStream: props.deliveryStreamNameAuditDestination,
+      };
+    }
+
+    var identifierArns: string[] = [];
+    if (props.identifiers != null) {
+      for (let identifier of props.identifiers) {
+        let identifierArn = Stack.of(scope).formatArn({
+          resource: 'data-identifier',
+          region: '',
+          account: 'aws',
+          service: 'dataprotection',
+          resourceName: identifier.toString(),
+        });
+        identifierArns.push(identifierArn);
+      };
+    }
+
+    if (props.identifierArnStrings != null) {
+      for (let identifierArnString of props.identifierArnStrings) {
+        identifierArns.push(identifierArnString);
+      }
+    }
+
+    this.statement = [
+      {
+        sid: 'audit-statement-cdk',
+        dataIdentifier: identifierArns,
+        operation: {
+          audit: {
+            findingsDestination: findingsDestination,
+          },
+        },
+      },
+      {
+        sid: 'redact-statement-cdk',
+        dataIdentifier: identifierArns,
+        operation: {
+          deidentify: {
+            maskConfig: {},
+          },
+        },
+      },
+    ];
+  }
+}
+
+type FindingsDestination = {
+  cloudWatchLogs?: CloudWatchLogsDestination;
+  firehose?: FirehoseDestination;
+  s3?: S3Destination;
+}
+
+type CloudWatchLogsDestination = {
+  logGroup: string;
+}
+
+type FirehoseDestination = {
+  deliveryStream: string;
+}
+
+type S3Destination = {
+  bucket: string;
+}
+
+/**
+ * Interface for creating a data protection policy
+ */
+export interface DataProtectionPolicyProps {
+  /**
+   * Name of the data protection policy
+   *
+   * @default - 'data-protection-policy-cdk'
+   */
+  readonly name?: string;
+
+  /**
+   * Description of the data protection policy
+   *
+   * @default - 'cdk generated data protection policy'
+   */
+  readonly description?: string;
+
+  /**
+   * List of data protection identifiers. Must be in the following list: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/protect-sensitive-log-data-types.html
+   *
+   * @default - no identifiers
+   */
+  readonly identifiers?: DataIdentifier[];
+
+  /**
+   * For futureproofing; if an identifier is not part of DataIdentifier, but still supported by data protection, a string (full ARN) can be supplied instead.
+   *
+   * @default - no string identifiers
+   */
+  readonly identifierArnStrings?: string[];
```

Comment:
Instead of doing this separately can we do it all in `identifiers` by changing
it to an enum like class? Something like

```ts
export DataIdentifier {
  public static ADDRESS = new DataIdentifier('Address');
  constructor(private readonly identifier: string) {}
  public toString(): string {
    return this.identifier;
  }
}
```

Then you can do `new DataIdentifier('somethig-new')`

Reply:
Changed.